### PR TITLE
refactor: moved to compatible with React 17

### PR DESCRIPTION
**Pull request by semoal**
- Changed peerDeps to react 15,16,17. **If we update react-transtion-group to ^3 | 4^ we must release a breaking change because drops compatibility with react 15**
- Removed karma,chai,enzyme blablabla... and used only Jest and Testing-Library.
- Updated devDependencies and cleared some.

Checks:
✅ npm run test
✅ npm run lint
✅ npm run demo -> 4 examples are correctly rendered

Questions:
❓ Should I create a Github Action file? Looks more interesting than Travis, that looks isn't working on pull requests. **I have already one ready if you're comfortable with github actions i push one**

### Master branch:
```
ReactSmooth.min.js  58.9 KiB       0  [emitted]  main
```

### This branch:
```
ReactSmooth.min.js  35.6 KiB       0  [emitted]  main
```

### CODE REVIEW

**Comment by semoal** [comment]
### Master branch:
```
ReactSmooth.min.js  58.9 KiB       0  [emitted]  main
```

### This branch:
```
ReactSmooth.min.js  35.6 KiB       0  [emitted]  main
```

**Comment by semoal** [comment]
@xile611 this pull request is ready, I can revert the lodash move if you don't feel "safe", i've tried this package in my projects and works good, but as you consider

**Comment by pm-mkorpar** [comment]
Any update on this? :)

File: package.json [hunk 1]
```diff
@@ -41,58 +41,45 @@
...
+    "@testing-library/dom": "^7.30.0",
+    "@testing-library/jest-dom": "^5.11.9",
+    "@testing-library/react": "^11.2.5",
     "babel-eslint": "^10.0.0",
     "babel-loader": "^8.0.0",
-    "babel-plugin-lodash": "^3.3.0",
-    "chai": "^4.1.2",
-    "chai-enzyme": "^0.8.0",
-    "chai-spies": "^0.7.1",
+    "core-js": "^3.9.1",
```
Comment (avindra) [inline package.json:71]: Unused?

Comment (semoal) [reply package.json:71]: What's unused? Corejs is required by presetenv

Comment (avindra) [reply package.json:71]: I gather it has to be explicitly enabled, and is disabled by default:

https://babeljs.io/docs/en/babel-preset-env#usebuiltins

(Which is good. Polyfill = application responsibility, not library)

Comment (semoal) [reply package.json:71]: That's not 100% right, on recharts we've been shipping library compiled with env to be compatible with at least ie11 and last 2 versions.

Comment (avindra) [reply package.json:71]: I noticed, and I think it's a bit unusual. The use of `Webpack` itself here is a bit unusual. It's a tool for bundling client js for applications. The underlying `babel` can/should be used by itself when we are creating libraries. Under the covers, webpack is just calling upon `babel` to do the transpile.

[axios](https://github.com/axios/axios) is another famous project for promulgating this approach, which has sharp edges.

How do large projects end up committing seemingly obvious errors?

Consider the timeline: in 2016 and earlier, we had very limited choices for putting together demos/docs. Naturally, `webpack` can do this for us, but some project(s) end up using webpack for the _build_ too, which is really better left for `babel` alone to do.

----

Seen from an application developers' perspective: "Oh, guess I picked up another `Promise` implementation." or "Here goes another way to do `Object.assign`."  The developer now needs to a) address bloat if there is a lot of polyfill cruft, b) worry about conflicts of interest in competing polyfill solutions.

The _application developer_ needs to be in charge of their target environment(s). If _they_ need a polyfill, they should add it...  we cannot possibly imagine all the possible environments the code could be running in. And it's simple enough for the consumer to just add what they need.

Comment (avindra) [reply package.json:71]: To get back on topic: @semoal

For this PR: I think it's fine if you need to add `core-js` (it was in recharts main repo at some point), but prefer to leave it out if possible.

I will write a more detailed post about this problem soon, as I haven't seen this problem described in any detail on the web. Sorry for the rant.

File: src/configUpdate.js [hunk 2]
```diff
@@ -65,7 +64,8 @@ export default (from, to, easing, duration, render) => {
   let update = () => null;
 
   const getCurrStyle = () => mapObject((key, val) => val.from, stepperStyle);
-  const shouldStopAnimation = () => !filter(stepperStyle, needContinue).length;
+  console.log(needContinue);
```
Comment (avindra) [inline src/configUpdate.js]: :scissors:

File: package.json [hunk 3]
```diff
@@ -41,58 +41,45 @@
...
   "homepage": "https://github.com/recharts/react-smooth#readme",
   "peerDependencies": {
-    "react": "^15.0.0 || ^16.0.0",
-    "react-dom": "^15.0.0 || ^16.0.0"
+    "react": "^15.0.0 || ^16.0.0 || ^17.0.0",
+    "react-dom": "^15.0.0 || ^16.0.0 || ^17.0.0",
+    "prop-types": "^15.6.0"
   },
   "dependencies": {
-    "lodash": "~4.17.4",
```
Comment (avindra) [inline package.json:48]: Nice :+1:

**Pull request merged**